packages/api/src/objectSet/ObjectSet.ts

```diff
@@ -254,4 +257,27 @@ export interface ObjectSet<
     listener: ObjectSetListener<Q, P>,
     opts?: ObjectSetListenerOptions<Q, P>,
   ) => { unsubscribe: () => void };
+
+  readonly withProperties: <
+    D extends DeriveClause<Q, any>,
+    P extends ObjectMetadata.Property,
```

Comment:
This generic isn't being captured for some reason, it will show if I type it as DeriveClause<Q,P> within the context of the derive clause, but it isn't being sent to the ExtendedObjectDefinition type